core/src/main/java/org/apache/accumulo/core/client/impl/TableMap.java

```diff
@@ -98,4 +106,8 @@ public TableMap(Instance instance, ZooCache zooCache) {
   public Map<String,String> getIdtoNameMap() {
     return tableIdToNameMap;
   }
+
+  public boolean isCurrent(ZooCache zc) {
+    return this.zooCache == zc && this.updateCount == zc.getUpdateCount();
+  }
```

Comment:
This method is nice.  Would it be better to use equals comparison?

Reply:
In this case I wanted to make sure it was the exact same zoocache instance

Reply:
Ah right, gotcha.